lib/bolt/cli.rb

```diff
@@ -379,710 +341,584 @@ def validate(options)
                 "Option '--env-var' can only be specified when running a command or script"
         end
       end
-    end
 
-    def handle_parser_errors
-      yield
-    rescue OptionParser::MissingArgument => e
-      raise Bolt::CLIError, "Option '#{e.args.first}' needs a parameter"
-    rescue OptionParser::InvalidArgument => e
-      raise Bolt::CLIError, "Invalid parameter specified for option '#{e.args.first}': #{e.args[1]}"
-    rescue OptionParser::InvalidOption, OptionParser::AmbiguousOption => e
-      raise Bolt::CLIError, "Unknown argument '#{e.args.first}'"
+      validate_targeting_options(options)
     end
 
-    def puppetdb_client
-      plugins.puppetdb_client
-    end
-
-    def plugins
-      @plugins ||= Bolt::Plugin.setup(config, pal, analytics)
-    end
-
-    def query_puppetdb_nodes(query)
-      puppetdb_client.query_certnames(query)
-    end
-
-    def warn_inventory_overrides_cli(opts)
-      inventory_source = if ENV[Bolt::Inventory::ENVIRONMENT_VAR]
-                           Bolt::Inventory::ENVIRONMENT_VAR
-                         elsif config.inventoryfile
-                           config.inventoryfile
-                         elsif File.exist?(config.default_inventoryfile)
-                           config.default_inventoryfile
-                         end
+    # Validates that only one targeting option is provided and that commands
+    # requiring a targeting option received one.
+    #
+    # @param options [Hash] The CLI options.
+    #
+    private def validate_targeting_options(options)
```

Comment:
This seems like it should return immediately for subcommands that don't take any targets, like `guide` and `group`, mostly because it seems odd to error if it's passed invalid options that aren't used.

Reply:
That would require having [this](https://github.com/puppetlabs/bolt/blob/main/lib/bolt/cli.rb#L444-L446), which I'm trying to avoid. Ideally, I'd like to have the parser just error when a command is given an invalid option, which theoretically can be done automatically by comparing its [flags](https://github.com/puppetlabs/bolt/blob/main/lib/bolt/bolt_option_parser.rb#L60) with the keys in the `options` hash. That'd be a "next step" of the refactor, though.